system_test/aest_hard_fork_SUITE.erl

```diff
@@ -642,6 +691,196 @@ new_node_can_mine_spend_tx_on_old_chain_using_new_protocol(Cfg) ->
     ?assertEqual(SpendTxBlockHash, maps:get(hash, get_block_by_height(new_node4, SpendTxHeight, Cfg))),
     ok.
 
+%% New node can sync the old chain from old node and can start mining
+%% on the top of the old chain. The new node can mine blocks using the
+%% old protocol and include contract create, contract call and contract
+%% call compute(?) transactions in the blocks. The blocks with the
+%% contract transactions can be synced among old and other new nodes.
+new_node_can_mine_contract_on_old_chain_using_old_protocol(Cfg) ->
+    {_, {_Saver, SavedCfg}} = proplists:lookup(saved_config, Cfg),
+    {_, old_node1} = proplists:lookup(old_node_left_running_with_old_chain, SavedCfg),
+    {_, new_node3} = proplists:lookup(new_node_left_running_with_synced_old_chain, SavedCfg),
+    {_, TopHeight} = proplists:lookup(db_backup_top_height, Cfg),
+    {_, _TopHash} = proplists:lookup(db_backup_top_hash, Cfg),
+    OldProtocolVersion = old_protocol_version(),
+    start_node(new_node4, Cfg),
+    wait_for_height_syncing(TopHeight, [new_node4], {{100000, ms}, {1000, blocks}}, Cfg),
+    ok = mock_pow_on_node(new_node3, Cfg),
+    ok = mock_pow_on_node(new_node4, Cfg),
+    ok = mock_pow_on_node(old_node1, Cfg),
+    %% expected_mine_rate is lowered so contract gets mined faster.
+    ErlCmd = make_erl_cmd("application:set_env(aecore, expected_mine_rate, ~p)", [10]),
+    ok = run_erl_cmd_on_node(new_node3, ErlCmd, Cfg),
+    ok = run_erl_cmd_on_node(new_node4, ErlCmd, Cfg),
+    ok = run_erl_cmd_on_node(old_node1, ErlCmd, Cfg),
+    %% Get public key of account meant to be creator of contract and make sure it has a sufficient balance by means of mining.
+    {ok, 200, #{pub_key := PubKey3}} = request(new_node3, 'GetPubKey', #{}, Cfg),
+    %% Balance is 0.
+    {ok, 404, #{reason := <<"Account not found">>}} =
+        request(new_node3, 'GetAccountBalance', #{account_pubkey => PubKey3}, Cfg),
+    ct:log("Balance of account with public key ~p is ~p", [PubKey3, 0]),
+    ok = run_erl_cmd_on_node(new_node3, "aec_conductor:start_mining().", 10000, Cfg),
```

Comment:
This is really no system test longer. In a system test, you can only use the system API.